### ISSUE TITLE
🐛 parse/semver: Unexpected word workspace

### ISSUE THREAD

**Issue by marudor** [romefrontend/rome#1076]
## Environment

```
$ rome rage --summary
Environment Variables
  Object {
    COLORTERM: "truecolor"
    LANG: "en_US.UTF-8"
    TERM: "xterm-256color"
    TERM_PROGRAM: "vscode"
  }

User Config
  unset

Rome Version
  10.0.4-beta

Node Version
  14.7.0

Platform
  darwin x64 19.6.0

Terminal Features
  Object {
    background: "unknown"
    colorDepth: 24
    columns: 187
    cursor: true
    hyperlinks: true
    isTTY: true
    unicode: true
  }

Client Flags
  Object {
    clientName: "cli"
    cwd: "/Users/marudor/programming/BahnhofsAbfahrten"
    realCwd: "/Users/marudor/programming/BahnhofsAbfahrten"
    silent: false
  }

Server Status
  Object {
    projects: Array [
      Object {id: 0}
      Object {id: 1}
    ]
    server: Object {
      heapTotal: 27_451_392
      pid: 98_870
      uptime: 0.173547076
    }
    workers: Array [
      Object {
        astCacheSize: 0
        heapTotal: 27_451_392
        ownedBytes: 0
        ownedFileCount: 0
        pid: 98_870
        uptime: 0.173482909
      }
    ]
  }
```

## Steps To Reproduce

1. have a workspace dependency in a monorepo (using yarn berry for instance)
Example dep:
```
"server": "workspace:*"
```
2. try to use rome

## Expected Results

It should recognize `workspace:*` as valid dependency

**Comment by sebmck** [comment]
It's not clear what we should do with these. Right now we are strict in how we parse `package.json` because later we will want to do many other things such as dependency installation. So it's better for us to iron out the bugs and require strict checks so we don't need to do it later.

We could possibly support the `workspace` pattern for compatibility, and later when we add dependency installation, just error on this.

The reason we would error on it is that Rome already implicitly supports workspaces without configuration. When we are initializing our file system, we basically have a single map of all packages inside of a project. That way, no matter where you are, you can `import` or `require` them. You might have noticed this since you didn't tell Rome about where your workspaces/packages were, but when linting, there weren't any dependency resolution errors because we just figured it out.

**Comment by marudor** [comment]
I would greatly appreciate to support the `workspace` pattern until dependency installation is there. That way workspace users can start using rome.

**Comment by sebmck** [comment]
Thinking about it more, it's reasonable. I'll get around to this later today as it's a relatively easy addition.

**Comment by sebmck** [comment]
Fixed by #1127.
